### DISCUSSION TITLE
Build a small artifact with `USE_SML_M` solve `Not enough space`

### DISCUSSION THREAD

**Discussion by deleonio** [arendst/Tasmota#19865] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff
- [x] Tasmota binary firmware version number used: 13.2
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: 
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:v
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:




```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
********************************************************************
* Looking for han_Parser.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:han_Parser.h"
* Web  > https://registry.platformio.org/search?q=header:han_Parser.h
*
********************************************************************

 #include "han_Parser.h"
                        ^
compilation terminated.
Retrieved `.pio/build/tasmota-ir/libbd8/ESP8266mDNS/ESP8266mDNS.cpp.o' from cache
*** [.pio/build/tasmota-ir/src/tasmota.ino.cpp.o] Error 1
====================================== [FAILED] Took 5.96 seconds ======================================

Environment    Status    Duration
-------------  --------  ------------
tasmota-ir     FAILED    00:00:05.958
================================ 1 failed, 0 succeeded in 00:00:05.958 ================================

```

### TO REPRODUCE
_Steps to reproduce the behavior: try to build `tasmota-ir`

> **Note**
> Your `master` with `SML` activated: https://github.com/deleonio/Tasmota/tree/23bb55bcd36b66afd4942c0c5cb1bf160b116e43

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
You did wrong, so happens as expected

![Screenshot_2023-10-29-09-28-43-725_com android chrome-edit](https://github.com/arendst/Tasmota/assets/2996042/6bfae919-e993-4be3-9894-e97c48b6fecd)

**Comment by deleonio** [comment]
Is there any way to make the default smaller. My device has only 1MB space and says `Not enough space`.

<img width="421" alt="image" src="https://github.com/arendst/Tasmota/assets/6279703/aec766d5-c238-4e55-88fd-2b213e885d32">

Thank you.

**Comment by barbudor** [comment]
Not related to your initial question

You should first upgrade to tasmota-minimal.bin.gz
The to your newly compiled tasmota.bin.gz

Take care of using directly the gz binaries

**Comment by deleonio** [comment]
I disabled some features in the `Default` variant. Ohjee 🙁 - but I have now a small artifact with SML.

Edit: `tasmota_configurations.h` -> https://github.com/deleonio/Tasmota/commit/05cf833f12f328e730f0d4bd60b3e45bd222509b#diff-00f571142d9f9ecce398802637a0fe4e073f1b2b6b36cd4712aba1a475c19567

<img width="404" alt="image" src="https://github.com/arendst/Tasmota/assets/6279703/b49c5d51-d399-429a-a907-6310ccb54870">

**Comment by barbudor** [comment]
When you start to modify Tasmota internal files instead of following the recommended way of working only with user_config_override.h and building tasmota/tasmota32, you are on your own

**Comment by deleonio** [comment]
What can I do then to enable SML and be able to create a very small artifact?

**Reply by barbudor** [reply]
In user_config_override.h add what is necessary to get SML
If you want to reduce the binary size you can #undef features that you don't need

But that shouldn't be necesseraly if you as told using Tasmota-minimal and the gz binaries

**Reply by barbudor** [reply]
If you don't want to go through compiling, you can take the ready binary from the unofficial binaries repository: https://github.com/tasmota/install/blob/firmware/firmware/unofficial/tasmota-scripting.bin.gz

Use the download button on the right:
![image](https://github.com/arendst/Tasmota/assets/2996042/4ed1a986-bbe0-402a-b0fb-71fbf04c8dca)

**Reply by barbudor** [reply]
And go through tasmota-minimal.bin.gz first : https://ota.tasmota.com/tasmota/tasmota-minimal.bin.gz
